src/neha_Patil/shop.java

```diff
@@ -0,0 +1,106 @@
+package neha_Patil;
+
+public class shop {
+
+
+	int stockofMaggie=00;
+	int stockofDosa=34;
+	int stockofOilPouches=43;
+	int stockofPanipuri=32;
+	int stockofmasala=71;
+
+	void requestedQ(boolean reqType,int maggieq, int dosaQ, int oilQ, int panipuriQ, int masalaQ ){ 
+		if(maggieq==0) 
+			System.out.println("Sorry maggie running out of stock");
+		else 
+			  if(stockofMaggie>=maggieq) {
+				System.out.println("We served your request for maggie");
+				stockofMaggie=stockofMaggie-maggieq;
+			  }
+				else if(reqType==true) {	
+					System.out.println("We are providing partial request of maggie "+maggieq);
+					maggieq=0;
+				}	
+
+		if(dosaQ==0) 
+			System.out.println("Sorry dosa running out of stock");
+		else if(stockofDosa>=dosaQ) {
+				System.out.println("We served your request for dosa packet");
+				stockofDosa=stockofDosa-dosaQ;
+		}
+				else if(reqType==true) {	
```

Comment:
You are already giving partial Dosa to customer in 
else if(stockofDosa>=dosaQ) {
				System.out.println("We served your request for dosa packet");
				**stockofDosa=stockofDosa-dosaQ;**
		}